qiskit/pulse/transforms.py

```diff
@@ -31,92 +31,103 @@ def align_measures(schedules: Iterable[interfaces.ScheduleComponent],
                    inst_map: Optional[InstructionScheduleMap] = None,
                    cal_gate: str = 'u3',
                    max_calibration_duration: Optional[int] = None,
-                   align_time: Optional[int] = None
+                   align_time: Optional[int] = None,
+                   align_all: Optional[bool] = True,
                    ) -> List[Schedule]:
-    """Return new schedules where measurements occur at the same physical time. Minimum measurement
-    wait time (to allow for calibration pulses) is enforced.
+    """Return new schedules where measurements occur at the same physical time.
+
+    Minimum measurement wait time (to allow for calibration pulses) is enforced
+    and may be set with ```max_calibration_duration`.
 
-    This is only defined for schedules that are acquire-less or acquire-final per channel: a
-    schedule with pulses or acquires occurring on a channel which has already had a measurement will
-    throw an error.
+    By default only instructions containing a :class:`~qiskit.pulse.AcquireChannel`
+    or :class:`~qiskit.pulse.MeasureChannel` will be shifted. If you wish to keep
+    the relative timing of all instructions in the schedule set ``align_all=True``.
+
+    This method assumes that ``MeasureChannel(i)`` and ``AcquireChannel(i)``
+    correspond to the same qubit and the acquire/play instructions
+    should be shifted together on these channels.
 
     Args:
         schedules: Collection of schedules to be aligned together
         inst_map: Mapping of circuit operations to pulse schedules
         cal_gate: The name of the gate to inspect for the calibration time
         max_calibration_duration: If provided, inst_map and cal_gate will be ignored
         align_time: If provided, this will be used as final align time.
-
+        align_all: Shift all instructions such that they happen at the same time
```

Comment:
This was not clear to me until I read the code (I think this option will control the time shift of non-measurement channels). Could you please add an example with jupyter execute?

Reply:
Added in e9ffa4f8b